### PR TITLE
Spatial partition design pattern

### PR DESCRIPTION
**Pull request by szoeke**
Spatial-partition first part. 
Still needs to work on it, but would like to make the first pull req.

### CODE REVIEW

**Comment by szoeke** [comment]
uploaded for review

File: spatial-partition/src/test/java/com/iluwatar/spatial/partition/SpatialPartitionTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    grid = new Grid();
+    unit = new Unit(grid, x, y);
+  }
+
+  @Test
+  public void initialiseEmpty() {
+    grid.clearGrid();
+    assertEquals(grid.getCells().length, 10);
+  }
+}
```
Comment (iluwatar) [inline spatial-partition/src/test/java/com/iluwatar/spatial/partition/SpatialPartitionTest.java]: The unit tests are insufficient. Add test for running the application. Add tests for `Grid` and `Unit` essential methods.

File: spatial-partition/src/main/java/com/iluwatar/spatial/partition/Unit.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+  public int getHealth() {
+    return health;
+  }
+
+  public void setHealth(int health) {
+    this.health = health;
+  }
+
+  void move(double x, double y) {
```
Comment (iluwatar) [inline spatial-partition/src/main/java/com/iluwatar/spatial/partition/Unit.java]: Should it be public method visibility?

File: spatial-partition/src/main/java/com/iluwatar/spatial/partition/Grid.java [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+        handleUnit(unit, cells[x][y - 1]);
+      }
+      if (x > 0 && y < NUM_CELLS - 1) {
+        handleUnit(unit, cells[x - 1][y + 1]);
+      }
+      unit = unit.getNextCell();
+    }
+  }
+
+  void moveUnitToNewCell(Unit unit, double x, double y) {
```
Comment (iluwatar) [inline spatial-partition/src/main/java/com/iluwatar/spatial/partition/Grid.java]: Should it be public method visibility?

**Comment by iluwatar** [comment]
@szoeke review done. Please notify when you've implemented the changes.

**Comment by iluwatar** [comment]
@szoeke this pull request is about to be closed due to inactivity. Please comment if you are still working on this.

**Comment by iluwatar** [comment]
Closed due to inactivity

**Pull request closed without merging**